Features.swift

```diff
@@ -8,9 +8,7 @@
 
 import Foundation
 
-#if canImport(CoreNFC)
```

Comment:
Several canImport(CoreNFC) have been removed here. Is this intentional? These were previously added because several online sources warned about this. If a phone does not have NFC capabilities, this could crash. As far as I know, the only non-capable NFC phone still supported by apple is iPhone 6s

Reply:
All iOS versions currently supported by Loop contain the CoreNFC framework, so the statement will always evaluate to true, during compile time.  At run time, the runtime NFC check on iPhone 6s will evaluate to false, which the code is checking as well.  The runtime check will also evaluate as false on the iOS simulator so you can test there to check behavior.

Reply:
@gui-dos This is a Loop Plugin, which only runs on iOS. I do think we should add NFC based bluetooth activation to this plugin; I have that working in an small test app that uses the LibreTools package: https://github.com/ivalkou/LibreTools, and it's relatively simple. That would make it convenient for users.

I imagine the first page of the pairing process in this plugin should be to scan the sensor. From there we know 1) whether the sensor is supported, 2) whether it needs activation, 3) whether it needs a transmitter addon like miaomiao or bubble, and can direct the user appropriately through activation and pairing. They would no longer need to navigate those steps manually.

Reply:
Each time you establish a Libre 2 BLE connection you have to increment an “unlock counter” by one. This is the simple reason why you loose the alarms coming from other apps when you scan a Libre 2 with a different device: when you send the NFC activation command  `A1 1E` such counter is reset to 0 and the sensor refuses to connect to the other apps sending the previous counter.

Luckily there are ways to bypass such limitations.

For example, @JohanDegraeve told me that he tested my code on his NFC-less iPhone SE 1 after activating the BLE streaming of data of his Libre 2s by using his wife’s iPhone and copying manually the original `patchInfo`, unlock code and counter parameters like I am allowing doing on an Apple Watch.

There is also a debug option in xDrip4iO5 (which I suggested B-) that lets you run the app in a companion mode since to decrypt the Libre 2 BLE data you don’t need the original `patchInfo` at the activation time but only the NFC unique identifier and you can decrypt the data without sending a BLE encoded unlock payload.

Notice that @ivalkou code implements only the decryption of the BLE data, not  how to encode such unlocking BLE payloads which I finished porting to Swift.

Unfortunately our Vendor is playing dirty with the new Libre 3: you have often to send the NFC commands multiple times and clean up the outputs from the varying dummy bytes `A5` they are prepending.

There is also a `receiverId` involved now which is a tied to a LibreView User Account ID therefore the sensor itself is capable of detecting whether it was already activated by the legitimate user…